_posts/2017-07-11-automated-security-testing.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: post
+title: Automated security testing
+author: lokori
+excerpt: How to implement a security pipeline and automate security testing? This post covers a PoC, based on open-source tools running inside Docker containers, which you can use as a template to implement your own solution.
+tags:  
+- DevSec
+- security
+- DevOps
+- DevSecOps
+- OWASP
+- Docker
+- ZAP
+- Security Pipeline
+---
+
+A Security Pipeline, or Automated security testing, is finally within our grasp! This is something we have been waiting for and trying to achieve for a number of years, but there's finally light at the end of the tunnel. This post provides a teplate for automating security scans and examines various issues one will encounter implementing that. You can find the low level details covered in this post at [Docker DevSec Demo](https://github.com/solita/docker-devsec-demo) repository in GitHub.
+
+## Simple: Build security in!
+
+The basic premise for Solita is that **security must be built in** during the development process. There is no security fairy who can magically add security to a product when it's almost finished. Dealing with serious security issues right before deployment to production is costly and often ruins the management's day by postponing the launch. Or makes them sweat if they have to launch with a system which is known to be vulnerable to hackers and cyber criminals. 
+
+We have previously covered this issue in high level at this previous post [What is DevSec](http://dev.solita.fi/2016/10/25/what-is-devsec.html). There was also a talk about this in [Tampere Goes Agile 2017](http://tamperegoesagile.fi/), [slides here](https://www.slideshare.net/Solita_Oy/devsec-build-security-in-and-dance-like-a-pro-81338146). 
+
+Our method is essentially a three step process:
+
+1. Ensure that the processes are reasonable and support making software secure.
+2. Do a threat analysis (with PO, customer and team all involved).
+3. Based on the treat analysis, create appropriate controls and plan security verification effort.
+
+Number three is where automated security testing can help. By automating certain parts of the security testing we can reduce risk and get better quality product for free. That's the idea, but of course in reality there are issues to deal with and still no silver bullet.
```

Comment:
The number three

and still no silver bullet. -> and still there is no silver bullet